common/src/main/java/io/pravega/common/util/CompositeBufferView.java

```diff
@@ -185,6 +201,69 @@ public int readBytes(ByteArraySegment segment) {
             return 0;
         }
 
+        @Override
+        public byte readByte() throws EOFException {
+            BufferView.Reader current = getCurrent();
+            if (current == null) {
+                throw new EOFException();
+            }
+
+            byte result = current.readByte();
+            this.available--;
+            assert this.available >= 0;
+            return result;
+        }
+
+        @Override
+        public int readInt() throws EOFException {
+            BufferView.Reader current = getCurrent();
+            if (current != null && current.available() >= Integer.BYTES) {
+                this.available -= Integer.BYTES;
+                assert this.available >= 0;
+                return current.readInt();
```

Comment:
Should the readInt call go before the update to available?

Reply:
I have already validated that `current.available()>4` so I am trying to avoid needing to declare another variable to hold the result so that I may decrement available here.

Reply:
I removed the assertion as it was superfluous.